boards/arm/stm32/stm32f4discovery/scripts/Make.defs

```diff
@@ -51,4 +51,4 @@ CELFFLAGS = $(CFLAGS) -mlong-calls # --target1-abs
 CXXELFFLAGS = $(CXXFLAGS) -mlong-calls # --target1-abs
 
 LDELFFLAGS = -r -e main
-LDELFFLAGS += -T $(call CONVERT_PATH,$(TOPDIR)/binfmt/libelf/gnu-elf.ld)
+LDELFFLAGS += -T $(call CONVERT_PATH,$(TOPDIR)/boards/arm/stm32/stm32f4discovery/scripts/gnu-elf.ld)
```

Comment:
@abhishek-samsung let's fix gnu-elf.ld directly.

Reply:
https://github.com/apache/nuttx/pull/11688#discussion_r1487851748

I was told not to direclty modify the gnu-elf.ld file, as it might be used for other architectures

Reply:
The nonexistent section doesn't have any bad side effect. @acassis is it good to only fix stm32f4discovery?

Reply:
@xiaoxiang781216 adding an ARM only section inside a generic libelf doesn't seams the right approach. An option is creating a generic section and at board level that section is defined and include the ARM specific section inside it (other boards from different arch could do the same). Maybe @patacongo could have a better solution.